GetIntoTeachingApi/Services/Store.cs

```diff
@@ -145,21 +151,23 @@ private async Task<IEnumerable<TeachingEvent>> OnlineEventsMatchingRequest(Teach
             return result.Where(te => te.IsOnline && !te.IsVirtual);
         }
 
+        private async Task SyncTeachingEventBuildings()
+        {
+            var buildings = _crm.GetTeachingEventBuildings();
+            await PopulateTeachingEventCoordinates(buildings);
+
+            await SyncModels(buildings, _dbContext.TeachingEventBuildings);
+        }
+
         private async Task SyncTeachingEvents()
         {
             var afterDate = DateTime.UtcNow.Subtract(TeachingEventArchiveSize);
             var teachingEvents = _crm.GetTeachingEvents(afterDate).ToList();
-            await PopulateTeachingEventCoordinates(teachingEvents);
-
-            var buildings = teachingEvents.Where(te => te.Building != null)
-                .Select(te => te.Building).DistinctBy(b => b.Id);
-
-            await SyncModels(buildings, _dbContext.TeachingEventBuildings);
+            var teachingEventBuildings = _crm.GetTeachingEventBuildings();

```

Comment:
This should load the buildings from the store as we've already cached them in the previous sync operation